packages/microsoft-security-adapter/src/client/oauth.ts

```diff
@@ -7,121 +7,53 @@
  */
 import axios from 'axios'
 import {
-  BuiltinTypes,
-  ElemID,
   InstanceElement,
   OAuthMethod,
   OAuthRequestParameters,
   OauthAccessTokenResponse,
   Values,
 } from '@salto-io/adapter-api'
-import { createMatchingObjectType } from '@salto-io/adapter-utils'
-import { ADAPTER_NAME } from '../constants'
-
-// While it’s true that at the moment we can maintain a single list of scopes for both adapters,
-// we do plan on enabling the user to choose to manage only Entra or Intune.
-// Therefore, it would be easier not to combine them for now.
-const ENTRA_OAUTH_REQUIRED_SCOPES = [
-  'AdministrativeUnit.ReadWrite.All',
-  'Application.ReadWrite.All',
-  'CustomSecAttributeDefinition.ReadWrite.All',
-  'Directory.ReadWrite.All',
-  'Domain.ReadWrite.All',
-  'Group.ReadWrite.All',
-  'Policy.Read.All',
-  'Policy.ReadWrite.AuthenticationMethod',
-  'Policy.ReadWrite.ConditionalAccess',
-  'Policy.ReadWrite.CrossTenantAccess',
-  'Policy.ReadWrite.PermissionGrant',
-  'RoleManagement.ReadWrite.Directory',
-  'UserAuthenticationMethod.ReadWrite.All',
-]
-
-export const INTUNE_OAUTH_REQUIRED_SCOPES = [
-  'DeviceManagementApps.ReadWrite.All',
-  'DeviceManagementConfiguration.ReadWrite.All',
-  'User.Read',
-]
-
-export const OAUTH_REQUIRED_SCOPES = [...new Set([...ENTRA_OAUTH_REQUIRED_SCOPES, ...INTUNE_OAUTH_REQUIRED_SCOPES])]
-
-export type OauthRequestParameters = {
-  tenantId: string
-  clientId: string
-  clientSecret: string
-  port: number
-}
+import {
+  AVAILABLE_MICROSOFT_SECURITY_SERVICES,
+  AvailableMicrosoftSecurityServices,
+  BASIC_OAUTH_REQUIRED_SCOPES,
+  Credentials,
+  MicrosoftServicesToManage,
+  SCOPE_MAPPING,
+} from '../auth'
 
-export const oauthRequestParameters = createMatchingObjectType<OauthRequestParameters>({
-  elemID: new ElemID(ADAPTER_NAME),
-  fields: {
-    tenantId: {
-      refType: BuiltinTypes.STRING,
-      annotations: {
-        message: 'Tenant ID',
-        _required: true,
-      },
-    },
-    clientId: {
-      refType: BuiltinTypes.STRING,
-      annotations: {
-        message: 'Client ID',
-        _required: true,
-      },
-    },
-    clientSecret: {
-      refType: BuiltinTypes.STRING,
-      annotations: {
-        message: 'Client Secret',
-        _required: true,
-      },
-    },
-    port: {
-      refType: BuiltinTypes.NUMBER,
-      annotations: {
-        message: 'Port',
-        _required: true,
-      },
-    },
-  },
-})
+const extractServicesToManageFromInputAsArray = (userInput: Values): AvailableMicrosoftSecurityServices[] =>
+  AVAILABLE_MICROSOFT_SECURITY_SERVICES.filter(service => userInput[service])
```

Comment:
nit: can we be explicit here?
```suggestion
  AVAILABLE_MICROSOFT_SECURITY_SERVICES.filter(service => userInput[service] === true)
```